src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -644,6 +748,21 @@ CAC.Map.Control = (function ($, Handlebars, cartodb, L, turf, _) {
         itineraries = {};
     }
 
+    /**
+     * Remove all itinerary map layers by inspecting for the GeoJSON `from` property.
+     *
+     * TODO: find better way to handle potential race conditions with live updating,
+     * which mean itinerary layers cannot be reliably removed by using the dictionary of
+     * itineraries stored in the map control during or on end of waypoint drag.
+     */
+    function removeItineraryLayers() {
+        map.eachLayer(function(layer) {
+            if (layer.feature && layer.feature.properties && layer.feature.properties.from) {
+                map.removeLayer(layer);
+            }
+        });
+    }
```

Comment:
Ah, I was thinking the problem was that we sometimes clobbered a layer reference before removing it.  But yeah, if the map has lost track and can't remove it by reference then keeping a copy isn't much help.